### PR TITLE
feat: Auto-Supports v1.5 - overhang mapping and placement fixes

### LINKED ISSUE #8: [TODO] Continue Island Volume Identification via Voxel-Based Logical Separation
### Task objective
Continue work on the voxel-based island volume identification system, with priority on finding and implementing an algorithm that can accurately and logically separate individual island volumes.

### Current problem (key note)
We **do not yet have an algorithm** that reliably performs logical island-volume identification.
The core blocker is accurately separating individual island volumes in a way that is both geometrically correct and practically usable downstream.

### Context
A voxel-based foundation already exists, but current approaches are not consistently producing correct logical separation of island volumes.

### Why this matters
This is critical for auto-support planning. If island volumes are misidentified (merged or split incorrectly), auto-support decisions become unreliable.

### Scope / boundaries
- Continue refining the existing voxel-based path.
- Focus on algorithm discovery/validation for logical volume separation.
- Keep output compatible with downstream auto-support systems.

### Technical focus areas
1. Connected-component + adjacency rules for volume segmentation.
2. Logical separation criteria for near-touching vs truly connected regions.
3. Threshold/tolerance handling by voxel resolution.
4. Deterministic output for repeatability.
5. Validation against known difficult models.

### Definition of done
1. A selected algorithm reliably identifies individual island volumes.
2. Known failing examples are correctly separated.
3. Output is stable enough for auto-support planning.
4. Regression tests cover split/merge edge cases and threshold behavior.

### Dependencies / blockers
- Final definition of “logically separate” for ambiguous geometry.
- Calibration of tolerances/resolution tradeoffs.

### LINKED ISSUE #384: [BUG] root supports do not get registered as supports
root supports are not registered as a support type, not rasterized, not exported, not destroyed.

Image

### LINKED ISSUE #473: [FEAT] Option to disable the Stability Warning dialog (horizontal angles in supports).
Image

### LINKED ISSUE #582: [BUG] Strange "conical effect" on mini supports
DragonFruit 0.1.13 (beta) — dev @ 1feaa0a4c · Windows/x64

"Mini supports" have a strange conical effect going from a size i cant seem to interact with to its tip dia. in contrast to just being a its set dia for the full range

Image

### LINKED ISSUE #585: Support connection balls not hidden while using the layer slider
If I slide through the layers in support mode the balls connecting the tips and trunks are always visible see picture. 
Tested in DragonFruit 0.1.13 (beta) — dev @ 1feaa0a4c · macOS/arm64

Image

### PR DESCRIPTION
**Pull request by PaulGD03**
## Summary

Second iteration of automatic support generation, rebuilt on what printing V1 outputs taught us. The distribution is now one fixed-density scheme (boundary ring plus grid infill per overhang region) instead of the V1 anchor-and-shape machinery, supports are sized from their own tier instead of the active trunk preset, and consolidated chunk trees make supports removable in pieces. Support Studio now loads the real parameters of a selected auto-generated support. This update also fixes the mid-air overhang contacts and the jaw support gaps seen on Puck.

This PR is a direct follow-up to #480 (Auto Supports V1). See the comparison section below.

---

### Distribution: one fixed-density scheme

- Every overhang region above the area threshold gets the same treatment: a boundary ring resampled at 0.7x infill spacing in the 2D-projected plane, plus a grid-lattice infill at the angle and suction spacing curve
- Surface resolution is triangle-accurate: containment and Z come from raycasts against the region's own triangles, with the voxel mask as fallback
- Shape handles the degenerate cases: slivers get a ring only, small patches a single pillar, per-region cap of 800 candidates with even subsampling
- The V1 machinery this replaces: anchor bands with lowest-Z-cluster selection, column tests, the competitive grid/Poisson bake-off, the Poisson disk generator, flatness dispatch, and the distribution mode selector. All were tried against real models and removed; the fixed-density scheme replaced them

### Chunk consolidation

- Neighbouring ring, infill and standalone trunks fan into each other so supports release in chunks (8 mm radius, 75 degree link angle, straight leaves only)
- Routed branches only for tips at 10 mm or higher; near the plate they read as a zig-zag web
- Chunk hosts match their fattest member. The per-attachment growth from V1 made uniform trees bulge at the attachment knots

### Decoupling from trunk presets

- Trunk presets in Support Studio are for manual placement only. Selecting one no longer changes auto-support density or sizing
- Auto-support sizing reads hardcoded tier bands (detail 0.8, structure 1.0, anchor 1.4 shafts) selected by the panel's Light/Medium/Heavy quick-select
- Auto-placed branches, leaves, twigs and sticks size from the same tier, so chunk trees are uniform-diameter

### Support Studio selection

- Auto-generated trunks, branches and leaves load their own parameters when selected. The settings stamp was previously encoded before tier overrides were applied, so the studio showed global parameters and any edit clobbered the sized geometry
- Added **Support Tooltips** toggle in General settings to hide `Cannot Place Support` and `Stability Warning` messages while placing supports

### Fixes carried on top of V1

- Twigs are cylindrical again; a span-based free-end taper rendered them as cones
- Connection balls are hidden while scrubbing layers in support mode
- Culled pillars report tip coordinates, and the report gained a consolidation-refusals line
- Orphan culling: trunkBlocked pillars are culled with reasons, legacy knots are rehosted, drift is caught after resize
- The density quick-select highlight no longer freezes on Light
- Anchors can no longer sink their shaft below the root joint: a near-plate tip lower than the root cone is rejected with a proper "Cannot Place Support" tooltip (new ANCHOR_BELOW_ROOT code) instead of building a cone that plunges into -Z, the cone length is clamped on downward axes so the socket lands on the joint, and the pathfinder (non-grid) hover preview now previews the anchor a click will actually place
- Anchors are now emitted into the slice mesh (root frustum, joint sphere, contact cone), so they rasterize in the layer preview and every sliced output
- Overhang island contacts sit on the surface: the contact combined the footprint bbox-centre XY with region.minZ, which floats mid-air below sloped or concave surfaces. It is now the lowest sampled pixel of the footprint mask with its real surface Z, so viewport pucks, camera focus, and auto-support tipPos all land on the model (#591)
- Elevated small overhangs (small islands or zHeight < 15) now route through the SmartPlacementV2 pathfinder instead of dropping as straight pillars that the post-resize cull deleted — fixes the missing supports under Puck's jaw and mouth
- Contact cones are clamped to their host shaft after the resize and merge passes (trunk top segment, leaf host segment, branch first segment), so a cone never renders wider than the shaft it sits on
- Fixed sideload world-transform for rust-scanned models: `load_and_transform` now replicates `processGeometry` double-centering (center X/Z, bottom Y) so `scan_islands_from_path` matches `prepareWorldGeom` — fixes sideloaded overhangs painting disconnected triangles without breaking the foot perimeter

### Diagnostics and reporting

- Forest Report now lists cavity fallbacks with tip positions and the fan refusal that preceded them (noHost, angle, blocked, etc.), and the cavity-fan is surfaced before bridging model-to-model
- Placement diagnostics surface fan, merge, consolidation and cavity counts in one place
- Forest Report modal widened to max-w-[72rem] to match the global Settings modal

### Cavity handling

- The model-to-model bridge fallback (stick/twig) now tries an 8 mm fan leaf on a nearby host before bridging — jaw tips that pierce the chest become regular tree leaves instead of visible sticks under the jaw
- Cavity sticks/twigs now raycast the shaft with trunk-matching clearance (radius + 0.15 mm) before placement — shafts that would shoot straight through the model are rejected instead of placed
- When the steepest host clips the mesh, the fan iterates to the next-best host by angle instead of giving up as blocked
- Long jaw to chest sticks (>12 mm bridge) are capped and rejected instead of placed, matching the older dev behaviour for isolated high overhangs

### Docs

- New auto-orientation pre-plan under the Developer Guide (`docs/dev/auto-orientation.md`): MSLA-tuned orientation scoring, with drainage scoring gated on hollowing state

### Cleanup

- Deleted anchorBands, distributionBakeoff and poissonPlacement with their suites
- Dev docs synced to the new architecture; superseded v1.5 findings removed
- Contact footprints packed into typed arrays (VoxelFootprint) and consumers ported

---

## Differences from V1 (#480)

| Area | V1 (#480) | V1.5 (this PR) |
| --- | --- | --- |
| Distribution | Shape-driven: grid on planar regions, Poisson disk on organic ones, with auto/grid/poisson modes | One fixed-density ring + grid infill scheme for every region; no modes |
| Anchoring | Anchor bands on the lowest Z-cluster, densified perimeter rings, anchor girth, anchors always standalone | No anchor concept; every region gets the same density, chunk trees handle removal |
| Anchor selection tuning | Bake-off between grid and Poisson per anchor, efficiency gates, column tests, spam guards | All removed; density alone decides |
| Sizing source | The active trunk preset (Support Studio) drove auto-support thickness | Auto supports have their own tier bands; trunk presets are manual-only |
| Chunk removal | Pillars were always 1:1, no consolidation | Neighbouring pillars consolidate into chunk trees |
| Twigs | Free end scaled with span (conical) | Cylindrical, both ends equal |
| Studio selection | Auto-generated supports loaded global parameters, edits clobbered sized geometry | Auto-generated supports load and edit their own parameters |
| Report | Refusal counters for fan/merge | Adds consolidation refusals, culled-pillar coordinates, cavity fallbacks with fan reasons, per-decision justification |

---

## Notes

- Known gap: pillars whose vertical column pierces the model (under arches, between legs) are still culled. Fixing that needs stepped landings, a separate feature

### Issue references

- Fixes #582 — conical mini supports (twigs with span-based taper rendered as cones)
- Fixes #585 — connection balls visible while scrubbing layers in support mode
- Closes #384 — root/anchor supports not rasterized into slices (the export/registration legs of that issue are tracked separately)
- Addresses #591 — floating overhang contacts (left open pending print-testing confirmation)
- Closes #473 — Support Tooltips toggle to disable Stability Warning and Cannot Place Support messages

### CODE REVIEW

File: src-tauri/src/mesh_minima.rs [hunk 1]
```diff
@@ -305,7 +355,7 @@ fn load_and_transform_mesh(
             pos.z = z;
         }
     }
-
+    let _ = center;
```
Comment (Jo-Con-El) [inline src-tauri/src/mesh_minima.rs:358]: ?
